src/libostree/ostree-fetcher.c

```diff
@@ -776,6 +759,13 @@ on_stream_read (GObject        *object,
                 gpointer        user_data);
 
 static void
+remove_pending_rerun_queue (OstreeFetcherPendingURI *pending)
+{
+  g_hash_table_remove (pending->thread_closure->outstanding, pending);
+  session_thread_process_pending_queue (pending->thread_closure);
```

Comment:
Better hold a reference on `pending` before removing it since that could free it and segfault on `pending->thread_closure`.

Also `s/pending_uri_free/pending_uri_unref/` please.